### ISSUE TITLE
Can I add HTTP proxy in python-vlc?

### ISSUE THREAD

**Issue by Darcy-C** [oaubert/python-vlc#168]
I found a HTTP proxy option in VLC, but I do not know how to set that in python-vlc.
Is there any docs about setting proxy?

Thank you very much.

**Comment by oaubert** [comment]
There is no proxy-related option in libvlc per se, but it should use the http_proxy environment variable if it is set. So you can `import os ; os.environ['http_proxy'] = 'http://whatever/'` before instanciating libvlc.

**Comment by Darcy-C** [comment]
Thank you for your reply.
It seems like not working when adding the lines you mentioned.

**Comment by mrJean1** [comment]
Try setting the proxy from the command line, like

`env  http_proxy='http://whatever...'  python  vlc.py ...`

Or perhaps use capitals like `env  HTTP_PROXY=....`

**Comment by Darcy-C** [comment]
> Try setting the proxy from the command line, like
> 
> `env http_proxy='http://whatever...' python vlc.py ...`
> 
> Or perhaps use capitals like `env HTTP_PROXY=....`

Thank you for your reply.
It does not work.
I thought it would be very hard to put a proxy setting to a third party library :c

**Comment by passgall** [comment]
> > Try setting the proxy from the command line, like
> > `env http_proxy='http://whatever...' python vlc.py ...`
> > Or perhaps use capitals like `env HTTP_PROXY=....`
> 
> Thank you for your reply.
> It does not work.
> I thought it would be very hard to put a proxy setting to a third party library :c

have you resolved this problem?

**Comment by Darcy-C** [comment]
> > > Try setting the proxy from the command line, like
> > > `env http_proxy='http://whatever...' python vlc.py ...`
> > > Or perhaps use capitals like `env HTTP_PROXY=....`
> > 
> > 
> > Thank you for your reply.
> > It does not work.
> > I thought it would be very hard to put a proxy setting to a third party library :c
> 
> have you resolved this problem?

I have not resolved this. Sorry.😥

**Comment by passgall** [comment]
> > > > Try setting the proxy from the command line, like
> > > > `env http_proxy='http://whatever...' python vlc.py ...`
> > > > Or perhaps use capitals like `env HTTP_PROXY=....`
> > > 
> > > 
> > > Thank you for your reply.
> > > It does not work.
> > > I thought it would be very hard to put a proxy setting to a third party library :c
> > 
> > 
> > have you resolved this problem?
> 
> I have not resolved this. Sorry.

If you have some other replace solutions? I want to view youtube videos with pyqt.I have not find other lib or tools which can embed in?Can you tell me your solution?

**Comment by Darcy-C** [comment]
> > > > > Try setting the proxy from the command line, like
> > > > > `env http_proxy='http://whatever...' python vlc.py ...`
> > > > > Or perhaps use capitals like `env HTTP_PROXY=....`
> > > > 
> > > > 
> > > > Thank you for your reply.
> > > > It does not work.
> > > > I thought it would be very hard to put a proxy setting to a third party library :c
> > > 
> > > 
> > > have you resolved this problem?
> > 
> > 
> > I have not resolved this. Sorry.
> 
> If you have some other replace solutions? I want to view youtube videos with pyqt.I have not find other lib or tools which can embed in?Can you tell me your solution?

Yes. If you are using pyqt, QtWebEngineWidgets is an option, which is just a browser, you can use web technology directly. I think it will fit your need very well.

**Comment by passgall** [comment]
> > > > > > Try setting the proxy from the command line, like
> > > > > > `env http_proxy='http://whatever...' python vlc.py ...`
> > > > > > Or perhaps use capitals like `env HTTP_PROXY=....`
> > > > > 
> > > > > 
> > > > > Thank you for your reply.
> > > > > It does not work.
> > > > > I thought it would be very hard to put a proxy setting to a third party library :c
> > > > 
> > > > 
> > > > have you resolved this problem?
> > > 
> > > 
> > > I have not resolved this. Sorry.
> > 
> > 
> > If you have some other replace solutions? I want to view youtube videos with pyqt.I have not find other lib or tools which can embed in?Can you tell me your solution?
> 
> Yes. If you are using pyqt, QtWebEngineWidgets is an option, which is just a browser, you can use web technology directly. I think it will fit your need very well.

maybe  python-mpv was a nice opiton! I have test it with proxy.

**Comment by vzhd1701** [comment]
The only way is to use the config file. `--http-proxy` command line option doesn't work, maybe because it is [grabbed directly from config](https://github.com/videolan/vlc/blob/5e70837d8d766db6ca5052a2d4f503ad37243d9c/src/win32/netconf.c#L36). Works for windows only though, in other systems VLC relies on OS proxy settings.

https://gist.github.com/vzhd1701/f8f8b2ee7b83baaa85fde4e37f02e3e6
